fhir-server/src/main/java/com/ibm/fhir/server/listener/FHIRServletContextListener.java

```diff
@@ -192,23 +199,88 @@ public void contextInitialized(ServletContextEvent event) {
             Boolean serverRegistryResourceProviderEnabled = fhirConfig.getBooleanProperty(PROPERTY_SERVER_REGISTRY_RESOURCE_PROVIDER_ENABLED, Boolean.FALSE);
             if (serverRegistryResourceProviderEnabled) {
                 log.info("Registering ServerRegistryResourceProvider...");
-                ServerRegistryResourceProvider provider = new ServerRegistryResourceProvider(persistenceHelper);
-                FHIRRegistry.getInstance().addProvider(provider);
-                FHIRPersistenceInterceptorMgr.getInstance().addInterceptor(provider);
+                FHIRRegistry.getInstance().addProvider(new ServerRegistryResourceProvider(persistenceHelper));
             }
 
-            Boolean graphTermServiceProviderEnabled = fhirConfig.getBooleanProperty(PROPERTY_GRAPH_TERM_SERVICE_PROVIDER_ENABLED, Boolean.FALSE);
-            if (graphTermServiceProviderEnabled) {
-                log.info("Adding GraphTermServiceProvider...");
-                PropertyGroup propertyGroup = fhirConfig.getPropertyGroup(PROPERTY_GRAPH_TERM_SERVICE_PROVIDER_CONFIGURATION);
-                if (propertyGroup == null) {
-                    log.log(Level.WARNING, "GraphTermServiceProvider configuration not found");
-                } else {
-                    Map<String, Object> map = new HashMap<>();
-                    propertyGroup.getProperties().stream().forEach(entry -> map.put(entry.getName(), entry.getValue()));
-                    int timeLimit = fhirConfig.getIntProperty(PROPERTY_GRAPH_TERM_SERVICE_PROVIDER_TIME_LIMIT, GraphTermServiceProvider.DEFAULT_TIME_LIMIT);
-                    graphTermServiceProvider = new GraphTermServiceProvider(new MapConfiguration(map), timeLimit);
-                    FHIRTermService.getInstance().addProvider(graphTermServiceProvider);
+            // Configure terminology service capabilities
```

Comment:
might be worth splitting this (or some of it) into a private method... this one is getting pretty long

Reply:
Will move this to a separate method per your suggestion.